source/dotnet/Library/AdaptiveCards/IgnoreNullEnumConverter.cs

```diff
@@ -0,0 +1,44 @@
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using System;
+using System.Collections.Generic;
+
+namespace AdaptiveCards
+{
+    public class IgnoreNullEnumConverter<TEnum> : StringEnumConverter, ILogWarnings
+    {
+        public List<AdaptiveWarning> Warnings { get; set; } = new List<AdaptiveWarning>();
+
+        // TODO: temporary warning code for invalid value. Remove when common set of error codes created and integrated.
+        private enum WarningStatusCode { UnknownElementType = 0 };
+
+        public IgnoreNullEnumConverter()
+        {
+        }
+
+        public IgnoreNullEnumConverter(bool camelCase) : base(camelCase)
+        {
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            try
+            {
+                // Try to read regularly
+                return base.ReadJson(reader, objectType, existingValue, serializer);
+            }
+            catch
+            {
+                // Catch invalid values and replace them with default value
+                // Add warning stating behavior
+                Warnings.Add(new AdaptiveWarning((int)WarningStatusCode.UnknownElementType, $"Value \"{reader.Value.ToString()}\" could not be converted to \"{typeof(TEnum).ToString()}\", using null instead."));
+                return null;
+            }
```

Comment:
does this duplicate what we do in the shared model? if a card has `"style": "someRandomString"`, should it be a warning or a parse error?

Reply:
I'm following the same pattern as the IgnoreDefaultStringEnumConverter, where a warning is also added, I assumed that's the expected behaviour

---
In reply to: [270215751](https://github.com/Microsoft/AdaptiveCards/pull/2580#discussion_r270215751) [](ancestors = 270215751)